pkg/subctl/cmd/root.go

```diff
@@ -234,3 +240,27 @@ var nodeLabelBackoff wait.Backoff = wait.Backoff{
 	Factor:   1.2,
 	Jitter:   1,
 }
+
+func checkVersionMismatch(cmd *cobra.Command, args []string) error {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("The provided kubeconfig is invalid", err)
+
+	clientSet, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner config", err)
+
+	submariner, err := clientSet.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if submariner != nil && submariner.Spec.Version != "" {
+		subctlVer, _ := goversion.NewVersion(version.Version)
+		submarinerVer, _ := goversion.NewVersion(submariner.Spec.Version)
+
+		if subctlVer != nil && submarinerVer != nil && subctlVer.LessThan(submarinerVer) {
+			return goerrors.New("subctl " + version.Version + " is too old. Please upgrade your version.")
```

Comment:
Maybe we should show the detected Submariner version here too.

Reply:
+1